lib/pub_sub/configuration.rb

```diff
@@ -41,9 +42,10 @@ def subscribe_to(service_name, messages: [])
     # You can specify the sender, messages, and topic - all independently.
     # If topic is not specified, the value for {service_identifier} will be used.
     def subscribe_to_custom(service_identifier, messages: [], topic: nil)
-      topic ||= service_identifier
       @subscriptions[service_identifier] = messages
-      @topics[service_identifier] = topic
+      @topics[service_identifier] = topic || service_identifier
+      # Assume handler class names based on conventions, ex: deal_update -> DealUpdate
+      @handlers = Hash[*messages.map { |m| [m, m.camelize.constantize] }.flatten]
```

Comment:
@ashoda Yes, and if they are incorrect at any point then the handlers won't be found when a message comes in.